napari/_qt/_tests/test_exception_handler.py

```diff
@@ -29,15 +32,50 @@ def test_keyboard_interupt_handler(qtbot, capsys):
     assert capsys.readouterr().err == "Closed by KeyboardInterrupt\n"
 
 
-def test_exception_handler_gui(qtbot, make_napari_viewer):
-    """Test exception handler can create a NapariNotification"""
-    viewer = make_napari_viewer()
-    handler = ExceptionHandler(gui_exceptions=True)
-    with qtbot.waitSignal(handler.error, timeout=1000):
-        handler.handle(ValueError, ValueError("whoops"), None)
-    assert handler.message in viewer.window.qt_viewer.canvas.native.children()
-    handler.message.toggle_expansion()
-    assert handler.message.property('expanded') is True
-    handler.message.toggle_expansion()
-    assert handler.message.property('expanded') is False
-    handler.message.close()
+# def test_exception_handler_gui(qtbot, make_napari_viewer):
+#     """Test exception handler can create a NapariQtNotification"""
+#     viewer = make_napari_viewer()
+#     handler = ExceptionHandler(gui_exceptions=True)
+#     with qtbot.waitSignal(handler.error, timeout=1000):
+#         handler.handle(ValueError, ValueError("whoops"), None)
+#     assert handler.message in viewer.window.qt_viewer.canvas.native.children()
+#     handler.message.toggle_expansion()
+#     assert handler.message.property('expanded') is True
+#     handler.message.toggle_expansion()
+#     assert handler.message.property('expanded') is False
+#     handler.message.close()
+
+
+def test_notification_manager(qtbot, make_napari_viewer):
+    """
+    Test that exception trigered by button in the UI, propagate to the manager,
+    and are displayed in the UI.
+    """
+
+    nothing = object()
+    orig = os.environ.get('DEBUG_NAPARI_NOTIFICATION', nothing)
+    os.environ['DEBUG_NAPARI_NOTIFICATION'] = '1'
+    try:
+        viewer = make_napari_viewer()
+        qtv = viewer.window.qt_viewer
+        errButton = qtv.layerButtons.newErrorButton
+        warnButton = qtv.layerButtons.newWarnButton
+        with notification_manager:
+            for btt, expected_message in [
+                (errButton, 'error!'),
+                (warnButton, 'warning!'),
+            ]:
+                assert btt is not None, errButton
+                assert len(notification_manager.records) == 0
+                qtbot.mouseClick(btt, Qt.LeftButton)
```

Comment:
I'm not too happy of having this test 1/2 checking Qt and 1/2 not; 
It looks like this is not yielding to the qt event loop so `viewer.children()` is not populated with the notifecation when I do so; any pointers ?

Reply:
hmm, i'll be honest we havn't been so great with the qtbot.mouseClick tests, we've generally just tested the callback, but i do like us doing more of this. maybe @tlambert03 or @goanpeca have an idea here?

Reply:
Yeah, I can help with this test... we really don't the button click here at all (it's just trigging an exception anyway).   I have a modified local test, but I'd like to address the static_method [here](https://github.com/napari/napari/pull/2205/files/7001cc25681ee44146950abcdebca31cad8b35a1#r583703165) first